### PR TITLE
Adds no-only-tests eslint plugin and rules

### PR DESCRIPTION
**Pull request by lanej0**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds the [no-only-tests](https://www.npmjs.com/package/eslint-plugin-no-only-tests) eslint plugin to ensure that focussed tests aren't pushed (breaking CI runs).

It's easy, when developing new tests or debugging existing tests for a `.only` to sneak through into production.  This plugin will ensure that that cannot happen.

### How to test the changes in this Pull Request:

1. Pull the branch
2. Open a test.  This should work for both Playwright tests (`e2e/tests/`) and Puppeteer tests (`tests/e2e/`)
3. Add `.only` to the end of either `it` or `test`.  It also works with `describe` blocks.
4. Save the change.  If your IDE is configured to run eslint on save, you should be warned of the violation.  If not, try to commit the change and it should be rejected in the pre-commit hook.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33572/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Changelog entry` f70ca9a3f8a72d61dffa9b70751e92b8e5c94dca
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

**Review by alopezari — DISMISSED** [review]
Thanks for creating this PR @lanej0! I approved it as all changes made sense and I could reproduce what's defined in the testing instructions.

Just out of curiosity, is there any particular reason why the webpack version was updated from `4.46.0_webpack-cli@3.3.12` to `4.46.0`?

**Comment by lanej0** [comment]
> Just out of curiosity, is there any particular reason why the webpack version was updated from 4.46.0_webpack-cli@3.3.12 to 4.46.0?

I think that was unintentional!  I'll double-check and remove that change before merge.

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -124,6 +124,7 @@ importers:
       jest: ^27.5.1
       jest-cli: ^27.5.1
       jest-mock-extended: ^1.0.18
+      require-turbo: workspace:*
```
Comment (lanej0) [inline pnpm-lock.yaml:127]: I rebased from trunk and re-installed the plugin.  These are changes from the original branch, but present in the current trunk.

**Comment by alopezari** [comment]
Hi @lanej0, thanks for updating this!

I spotted that when we place the `.only` option to a playwright test and try to commit that change, the following appears in the terminal:

```
/Users/alopezari/Projects/WooCommerce/woocommerce/plugins/woocommerce/e2e/tests/merchant/order-edit.spec.js
  36:7  error  Unexpected focused test  jest/no-focused-tests
  36:7  error  test.only not permitted  no-only-tests/no-only-tests
```

However, when we do the same with a puppeteer test, the following appears:
```
/Users/alopezari/Projects/WooCommerce/woocommerce/packages/js/e2e-core-tests/src/specs/merchant/wp-admin-order-edit.test.js
   32:6  error    Unexpected focused test         jest/no-focused-tests
```

Perhaps I didn't understand properly the scope of the changes in the previous PR review, as I thought that the `Unexpected focused test         jest/no-focused-tests` check was added here, but it was already in place, isn't it? This PR is for adding the `test.only not permitted  no-only-tests/no-only-tests` instead.

Seems this only applies to playwright tests, but not puppeteer ones, is this correct?. 
Also, seems that `no-only-tests` does the same as `no-focused-tests` in this context if I'm not wrong. Do we need to add `no-only-tests` then?

**Comment by lanej0** [comment]
After merging in the Playwright PR, I'm seeing a warning for focussed tests that wasn't there before.  So I think something changed, and this PR is no longer necessary.  Thanks for all of your reviews on this @alopezari

**Pull request closed without merging**